src/edit_session.js

```diff
@@ -187,6 +191,35 @@ class EditSession {
         return this.doc;
     }
 
+    /**
+     * Get "widgetManager" from EditSession
+     * 
+     * @returns {LineWidgets} object
+     */
+    get widgetManager() {
+        if(!this.$widgetManager) {
+        this.$widgetManager = new LineWidgets(this);
+        
+            if (this.$editor)
+                this.$widgetManager.attach(this.$editor);
+            }
+       return this.$widgetManager;
+    }
+
+    /**
+     * Set "widgetManager" in EditSession
+     * 
+     * @returns void
+     */
+    set widgetManager(value) {
+        Object.defineProperty(this, "widgetManager", {
```

Comment:
This code was intended for a use case of shadowing widgetManager getter from prototype with a property defined directly on the session object, so this conflicts with the approach of using `$widgetManager` property. 

Did you change the approach because  you think shadowing has worse performance and forgot to remove this, or did you make an error in adding the property with $?

Reply:
$widgetManager added this property so that we can store the widgetManger and return the same widget manager when called session.widgetManager multiple times.

Reply:
with this defineProperty call, it works without  $widgetManager  property

Reply:
LineWidgets constructor already does

https://github.com/ajaxorg/ace/blob/5e1e524d7ef04e13291b5a979fb4166e973e61f0/src/line_widgets.js#L18

so the defineProperty call in setter is enough to simply use getter and always get the same value